sharktank/sharktank/ops/default_impls.py

```diff
@@ -500,16 +500,33 @@ def scatter__default(
     *,
     reduce: str | None = None,
 ) -> Tensor:
-    assert isinstance(value, Number), "Tensor version of this op not implemented"
     inout = unbox_tensor(inout)
     index = unbox_tensor(index)
+    if isinstance(value, (torch.Tensor, InferenceTensor)):
+        assert isinstance(value, (torch.Tensor, PrimitiveTensor))
```

Comment:
How come this isn't being handled by the override and the trampoline?

Reply:
We don't dispatch on value.

Reply:
`torch.scatter_add` actually does not support source as number, but `torch.scatter` does. I refactored `scatter_` to dispatch on the source as well.
Had to fix the trivially replicable dispatch logic to also handle non-tensor dispatch args.